runtime/src/append_vec.rs

```diff
@@ -806,7 +815,7 @@ pub mod tests {
             // Depending on use, *executable_bool can be truthy or falsy due to direct memory manipulation
             // assert_eq! thinks *exeutable_bool is equal to false but the if condition thinks it's not, contradictly.
             assert_eq!(*executable_bool, false);
-            if *executable_bool == false {
+            if !*executable_bool {
```

Comment:
@svenski123 Well, some kind of dark-magic is happening here. Could you `#[allow(...)]` and revert this only line instead of changing the code? Well, this test failed on ci. :) https://buildkite.com/solana-labs/solana/builds/24275#d60807c1-87fd-4868-b7fc-c1f759ed9b29/123-1374

If interested, you can see the comments. :)

Reply:
Thanks for the heads up - I've made that test clippy compliant (and run tests on runtime + clippy --tests).

Rust bool handling is quite surprising / disturing - this assertion fails for values other than 0 or 1 (but still says left and right are both true):
```
let x: u8 = 0xcf; // can be anything > 1
let y: bool = unsafe { *(&x as *const u8 as *const bool) };
assert_eq!(y == false, if y == false { true } else { false });
```
Yes it's undefined behaviour in rust but even still - the expression evaluates differently as an if condition rather than standalone. Perhaps rust does a SETZ / JZ for the if condition and additional bit masking otherwise but ... yuck.